plugins/woocommerce/src/Internal/Logging/RemoteLogger.php

```diff
@@ -212,46 +221,41 @@ protected function should_handle( $level, $message, $context ) {
 	 * @return bool
 	 */
 	private function log( $level, $message, $context ) {
-		try {
-			$log_data = $this->get_formatted_log( $level, $message, $context );
+		$log_data = $this->get_formatted_log( $level, $message, $context );
 
 			// Ensure the log data is valid.
-			if ( ! is_array( $log_data ) || empty( $log_data['message'] ) || empty( $log_data['feature'] ) ) {
-				return false;
-			}
+		if ( ! is_array( $log_data ) || empty( $log_data['message'] ) || empty( $log_data['feature'] ) ) {
+			return false;
+		}
 
-			$body = array(
-				'params' => wp_json_encode( $log_data ),
-			);
+		$body = array(
+			'params' => SafeGlobalFunctionProxy::wp_json_encode( $log_data ) ?? 'Error occurred while encoding the log data',
+		);
 
-			WC_Rate_Limiter::set_rate_limit( self::RATE_LIMIT_ID, self::RATE_LIMIT_DELAY );
+		WC_Rate_Limiter::set_rate_limit( self::RATE_LIMIT_ID, self::RATE_LIMIT_DELAY );
 
-			if ( $this->is_dev_or_local_environment() ) {
-				return false;
-			}
+		if ( $this->is_dev_or_local_environment() ) {
+			return false;
+		}
 
-			$response = wp_safe_remote_post(
-				self::LOG_ENDPOINT,
-				array(
-					'body'     => wp_json_encode( $body ),
-					'timeout'  => 3,
-					'headers'  => array(
-						'Content-Type' => 'application/json',
-					),
-					'blocking' => false,
-				)
-			);
-
-			if ( is_wp_error( $response ) ) {
-				throw new \Exception( $response->get_error_message() );
-			}
+		$response = SafeGlobalFunctionProxy::wp_safe_remote_post(
+			self::LOG_ENDPOINT,
+			array(
+				'body'     => SafeGlobalFunctionProxy::wp_json_encode( $body ) ?? 'Error occurred while encoding the log data',
+				'timeout'  => 3,
+				'headers'  => array(
+					'Content-Type' => 'application/json',
+				),
+				'blocking' => true,
+			)
+		);
 
-			return true;
-		} catch ( \Throwable $e ) {
-			// Log the error locally if the remote logging fails.
-			error_log( 'Remote logging failed: ' . $e->getMessage() ); // phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log
+		if ( SafeGlobalFunctionProxy::is_wp_error( $response ) ) {
```

Comment:
Should we throw an exception here when unable to call the is_wp_error function since we can't determine if the response is an error or not?

Reply:
Good call! I think testing for null here should suffice - that's what it'll return if the call fails. What do you think?

Reply:
Sounds good!